changelog.d/10969.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug where messages in `device_inbox` table where not deleted, when deleting device(s).
```

Comment:
I'm guessing this has been around basically forever, so I suggest we describe the bug `long-standing`. (Unless we happen to know that it was introduced in a particular release?)

I've tweaked the wording a little too, to emphasise the pain for the server administrator (lots of DB rows!). I've also added a sentence to attribute yourself and Johannes. What do you think?

```suggestion
Fix a long-standing bug where messages in the `device_inbox` table for deleted devices would persist indefinitely. Contributed by @dklimpel and @JohannesKleine.
```

Reply:
I think this is a bug since the beginning.